File: Patrones/Bridge/C++/BridgeStackSimple/src/StackHanoi.cpp
#include <StackHanoi.h>

StackHanoi::StackHanoi() : Stack("array"){
	totalRejected = 0;	
}

StackHanoi::StackHanoi(string s) : Stack(s){
	totalRejected = 0;
}

int StackHanoi::reportRejected(){
	return totalRejected;
}

void StackHanoi::push( int in ){
    if ( !isEmpty()  &&  in > top()) {
        totalRejected++;        
    }
    else {
        Stack::push( in );
    }
}

StackHanoi::~StackHanoi(){}

File: Patrones/Bridge/Java/BridgeStackSimple/src/example/StackHanoi.java
package example;

//Embellish the interface class with derived classes if desired
public class StackHanoi extends Stack {
	
	private int totalRejected = 0;
	
	public StackHanoi(){ 
		super( "java" ); 
	}
	
	public StackHanoi( String s ){
		super( s );
	}
	
	public int reportRejected(){
		return totalRejected;
	}
	
	public void push( int in ){
		if ( !imp.empty()  &&  in > ((Integer)imp.peek()).intValue())
			totalRejected++;
	    else
	    	imp.push( new Integer(in) );
	}
}
